### PR TITLE
Only run the necessary steps in parallel.

### PR DESCRIPTION
**Pull request by beverloo**
This addresses @foolip's #9

### CODE REVIEW

**Comment by beverloo** [comment]
@foolip - mind taking a quick peek? Thank you for the feedback!

File: index.bs [hunk 1]
```diff
@@ -292,20 +292,27 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
...
     <ol>
       <li>
+        Let <var>promise</var> be <a>a new promise</a>.
+      </li>
+      <li>
         Let <var>origin</var> be the <a>entry settings object</a>'s <a>origin</a>.
       </li>
       <li>
         If the <var>origin</var> is not a <a>secure context</a>, reject <var>promise</var> with a
         <code>{{SecurityError}}</code> and terminate these steps.
```
Comment (foolip) [inline index.bs:306]: Unrelated to the problem I reported, but in this code path the promise isn't actually returned. s/and terminate these steps/and return promise/. The new deal in https://infra.spec.whatwg.org/#algorithm-control-flow is that "return" implies "terminate these steps".

File: index.bs [hunk 2]
```diff
@@ -315,62 +322,69 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
...
```
Comment (foolip) [inline index.bs:336]: Need to return here too. I won't look for or comment on more instances.

File: index.bs [hunk 3]
```diff
@@ -315,62 +322,69 @@ spec: webidl; urlPrefix: https://heycam.github.io/webidl/
...
-      </li>
-      <li>
-        If there are entries in <var>origin</var>'s <a>list of budget expectations</a>, for each
-        <var>entry</var>:
-
+        Return <var>promise</var> and run the following step <a>in parallel</a>:
         <ol>
           <li>
-            Let <var>state</var> be a new <code>{{BudgetState}}</code> instance.
+            Let <var>details</var> be a new <code>{{sequence}}</code>.
```
Comment (foolip) [inline index.bs:342]: Also unrelated, but using https://infra.spec.whatwg.org/#list would be more idiomatic I think, and Web IDL says "Any list can be implicitly treated as a sequence<T>, as long as it contains only items that are of type T." while linking to infra.

**Review by foolip — APPROVED** [review]
Some unrelated nits, may fix separately?

**Pull request merged**